### PR TITLE
cmd/ls: Add ls command

### PR DESCRIPTION
**Pull request by Prnyself**
cmd/ls: Add ls command, bucket acl interface and relevant convert, align func for printing

### CODE REVIEW

File: storage/qingstor.go [hunk 1]
```diff
@@ -214,3 +215,85 @@ func (q *QingStorObjectStorage) DeleteObject(objectKey string) (err error) {
...
+			convert.Int64Value(obj.Size),
+			convert.StringValue(obj.MimeType),
+			convert.StringValue(obj.Etag),
+			time.Unix(int64(convert.IntValue(obj.Modified)), 0),
+			convert.StringValue(obj.StorageClass),
+		})
+	}
+
+	// recursively for next marker request
+	if convert.BoolValue(res.HasMore) {
```
Comment (Xuanwo) [inline storage/qingstor.go]: Let's use `for` directly, do not call `q.ListObjects`

File: storage/qingstor.go [hunk 2]
```diff
@@ -214,3 +215,85 @@ func (q *QingStorObjectStorage) DeleteObject(objectKey string) (err error) {
...
+		}
+		oms = append(oms, rOms...)
+	}
+	return
+}
+
+// GetBucketACL will get acl from a bucket.
+func (q *QingStorObjectStorage) GetBucketACL() (ar *ACLResp, err error) {
+	res, err := q.bucket.GetACL()
+	if err != nil {
```
Comment (Xuanwo) [inline storage/qingstor.go]: Print error message here.

File: storage/mock.go [hunk 3]
```diff
@@ -163,3 +163,18 @@ func (m *MockObjectStorage) DeleteObject(objectKey string) (err error) {
...
 	return nil
 }
+
+// ListBuckets implements ObjectStorage.ListBuckets
+func (m *MockObjectStorage) ListBuckets(zone string) (buckets []string, err error) {
+	return []string{}, nil
+}
+
+// ListObjects implements ObjectStorage.ListObjects
+func (m *MockObjectStorage) ListObjects(prefix, delimiter string, marker *string) (om []*ObjectMeta, err error) {
```
Comment (Xuanwo) [inline storage/mock.go]: We need to return fake data for test.

File: utils/print_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	assert.Equal(t, len(ori), 3)
+
+	res := make([]string, len(ori))
+	for i, line := range ori {
+		res[i] = strings.Join(line, " ")
+		fmt.Println(res[i])
+	}
+
+	expected :=
```
Comment (Xuanwo) [inline utils/print_test.go]: We can use

```
x := `xxx
xxxx
`
```

Comment (Prnyself) [reply utils/print_test.go]: > We can use
> 
> ```
> x := `xxx
> xxxx
> `
> ```

I prefer plus each part of string for a better source-code-read format.

Comment (Xuanwo) [reply utils/print_test.go]: ok

File: storage/qingstor.go (hunk 2 above)
Comment (Prnyself) [reply storage/qingstor.go]: done by next commit

File: constants/time.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package constants
+
+// Supported time layouts
+const (
+	RFC822       = "Mon, 02 Jan 2006 15:04:05 GMT"
+	ISO8601      = "2006-01-02T15:04:05Z"
+	ISO8601Milli = "2006-01-02T15:04:05.000Z"
+	NGINXTime    = "02/Jan/2006:15:04:05 -0700"
+	CTLCustom    = "Jan 02 15:04"
```
[the remark this answers is not in the document]
Comment (Prnyself) [reply constants/time.go]: modified to LsDefaultFormat for better implication

File: storage/qingstor.go (hunk 1 above)
Comment (Prnyself) [reply storage/qingstor.go]: done by next commit

File: storage/mock.go [hunk 6]
```diff
@@ -19,24 +20,40 @@ const (
 	MockMBObject = "mb"
 	MockGBObject = "gb"
 	MockTBObject = "tb"
+
+	MockPek3a = "mock-pek3a"
```
Comment (Xuanwo) [inline storage/mock.go]: We do not use valid zone name in mock, instead, we can use `MockZoneAlpha`, `MockZoneBeta`.

And two zone is enough.

**Pull request merged**